### PR TITLE
Adjust and refactor all `FlatButton` tests into its respective file

### PR DESCRIPTION
**Pull request by Inconnu08**
## Description

The existing monolithic `buttons_tests.dart` file contains tests for different buttons like `FlatButton`, `RaisedButton`, `OutlineButton` and `MaterialButton`, even though `flat_button_test.dart`, `raised_button_test.dart`, `outline_button_test.dart`, and `material_button_test.dart` exists.

This PR splits up and adjusts `FlatButton` tests properly into its respective file.

## Related Issues

 #42798

## Tests

I added the following tests:

- Refactors existing

## Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/test/material/flat_button_test.dart [hunk 1]
```diff
@@ -334,79 +412,209 @@ void main() {
     );
   });
 
-  testWidgets('FlatButton onPressed and onLongPress callbacks are correctly called when non-null', (WidgetTester tester) async {
+  testWidgets('Does FlatButton work with hover', (WidgetTester tester) async {
```
Comment (shihaohong) [inline packages/flutter/test/material/flat_button_test.dart]: It looks like you might have deleted two newer preexisting tests on `flat_button_test.dart` by mistake: 
- 'FlatButton onPressed and onLongPress callbacks are correctly called when non-null'
- 'FlatButton onPressed and onLongPress callbacks are distinctly recognized'

File: packages/flutter/test/material/flat_button_test.dart [hunk 2]
```diff
@@ -334,79 +412,209 @@ void main() {
...
+      ),
     );
-    flatButton = find.byType(FlatButton);
-    expect(tester.widget<FlatButton>(flatButton).enabled, false);
-  });
 
-  testWidgets('FlatButton onPressed and onLongPress callbacks are distinctly recognized', (WidgetTester tester) async {
-    bool didPressButton = false;
-    bool didLongPressButton = false;
+    expect(tester.getSize(find.byType(FlatButton)), equals(const Size(88.0, 48.0)));
```
Comment (shihaohong) [inline packages/flutter/test/material/flat_button_test.dart]: It looks like you might have deleted two newer preexisting tests on `flat_button_test.dart` by mistake: 
- 'FlatButton onPressed and onLongPress callbacks are correctly called when non-null'
- 'FlatButton onPressed and onLongPress callbacks are distinctly recognized'

File: packages/flutter/test/material/flat_button_test.dart (hunk 1 above)
Comment (Inconnu08) [reply packages/flutter/test/material/flat_button_test.dart]: Ah, sorry. Accidentally messed up during a merge conflict, but restored it now

File: packages/flutter/test/material/flat_button_test.dart [hunk 3]
```diff
@@ -377,7 +659,7 @@ void main() {
     flatButton = find.byType(FlatButton);
     expect(tester.widget<FlatButton>(flatButton).enabled, false);
   });
-
+
```
Comment (shihaohong) [inline packages/flutter/test/material/flat_button_test.dart]: Causes analyzer whitespace problem
```suggestion

```

**Review by shihaohong — APPROVED** [review]
LGTM. Just make sure to fix pre-submit tests!

**Pull request merged**